src/crawlers/browser_crawler.js

```diff
@@ -374,12 +375,7 @@ class BrowserCrawler extends BasicCrawler {
 
             if (session) session.markGood();
         } finally {
-            try {
-                await page.close();
-            } catch (error) {
-                // Only log error in page close.
-                this.log.debug('Error while closing page', { error });
-            }
+            await page.close().catch((error) => this.log.debug('Error while closing page', { error }));
```

Comment:
Don't await it. That's the point of the `close().catch()` instead of `try/catch`.

It prevents opening of a new page (running a new task) before this one fully closes. We don't need to wait for the page to close.